otlp-logger.js

```diff
@@ -0,0 +1,153 @@
+'use strict'
+
+const {
+  LoggerProvider,
+  SimpleLogRecordProcessor,
+  BatchLogRecordProcessor
+} = require('@opentelemetry/sdk-logs')
+
+const { SeverityNumber } = require('@opentelemetry/api-logs')
+const { logs } = require('@opentelemetry/api-logs')
+const { Resource } = require('@opentelemetry/resources')
+
+const DEFAULT_MESSAGE_KEY = 'msg'
+
+/**
+ * @typedef {Object} Options
+ * @property {string} loggerName
+ * @property {string} serviceName
+ * @property {string} serviceVersion
+ * @property {boolean} includeTraceContext
+ * @property {import('@opentelemetry/sdk-logs').LogRecordExporter} [logRecordExporter]
+ * @property {boolean} [useBatchProcessor=true]
+ * @property {string} [messageKey="msg"]
+ *
+ * @param {Options} opts
+ */
+function getOtlpLogger (opts) {
+  const loggerProvider = new LoggerProvider({
+    resource: new Resource({
+      'service.name': opts.serviceName,
+      'service.version': opts.serviceVersion
+    })
+  })
+
+  const recordProcessor =
+    opts.useBatchProcessor ?? true
+      ? new BatchLogRecordProcessor(opts.logRecordExporter)
+      : new SimpleLogRecordProcessor(opts.logRecordExporter)
+
+  loggerProvider.addLogRecordProcessor(recordProcessor)
```

Comment:
What if someone wants multiple exporters? e.g. `ConsoleLogRecordExporter` & `OTLPLogExporter`

Reply:
That's a good point. I couldn't think of a simple API that would support various combinations of exporters (GRPC, HTTP, Console, InMemory ...) and processors(Batch, simple ...). Do you have some suggestions? I'll create a dedicated issue to discuss and address this.

Reply:
https://github.com/Vunovati/pino-opentelemetry-transport/issues/3